### ISSUE TITLE
Request list of services if service list is empty

### ISSUE THREAD

**Issue by objectiser** [jaegertracing/jaeger-ui#496]
## Requirement - what kind of business use case are you trying to solve?
See services in the dropdown that have reported traces after the UI had already started.

## Problem - what in Jaeger blocks you from solving the requirement?
When starting up a Jaeger UI in dev/test environment, if no traces have been reported the service dropdown is empty. When traces are reported, the user is required to refresh the browser to see the actual list.

## Proposal - what do you suggest to solve the problem or improve the existing situation?
If the service list is empty, then actively query the backend to retrieve the initial list.

## Any open questions to address
We may also want to consider a configurable polling period to automatically update the cached service list.

**Comment by yurishkuro** [comment]
Good idea. I would use the count in parentheses as a progress indicator, ie replace 0 with "loading..." while loading.

What did you mean by cached service list? Afaik UI does not cache it; it's loaded every time you visit the Search page.

**Comment by objectiser** [comment]
> What did you mean by cached service list?

My assumption - I normally just refresh the browser to get an uptodate list - but I guess changing between tabs might work. The main thing is that (ideally) it would be nice if the user does not need to necessarily force that update to see the latest list.

**Comment by yurishkuro** [comment]
right, so `state` rather than `cache` of the UI.
